beacon-chain/state/stateutil/attestations.go

```diff
@@ -88,6 +88,64 @@ func marshalAttestationData(data *ethpb.AttestationData) []byte {
 	return enc
 }
 
+func attestationRoot(att *ethpb.Attestation) ([32]byte, error) {
+	fieldRoots := make([][32]byte, 3)
+
+	// Bitfield.
+	aggregationRoot, err := bitlistRoot(att.AggregationBits, 2048)
+	if err != nil {
+		return [32]byte{}, err
+	}
+	fieldRoots[0] = aggregationRoot
+
+	dataRoot, err := attestationDataRoot(att.Data)
+	if err != nil {
+		return [32]byte{}, err
+	}
+	fieldRoots[1] = dataRoot
+
+	signatureBuf := bytesutil.ToBytes96(att.Signature)
+	packedSig, err := pack([][]byte{signatureBuf[:]})
+	if err != nil {
+		return [32]byte{}, err
+	}
+	sigRoot, err := bitwiseMerkleize(packedSig, uint64(len(packedSig)), uint64(len(packedSig)))
+	if err != nil {
+		return [32]byte{}, err
+	}
+	fieldRoots[2] = sigRoot
+	return bitwiseMerkleizeArrays(fieldRoots, uint64(len(fieldRoots)), uint64(len(fieldRoots)))
+}
+
+func blockAttestationRoot(atts []*ethpb.Attestation) ([32]byte, error) {
```

Comment:
Prefer naming it `attestationsRoot` as it doesn't need to be constrained with `block`, the function can be extensible to any object that needs a hash tree root of a list of attestations

Reply:
we use the max block attestation capacity here, so this will only apply for beacon block attestations. If we use attestations which are larger than 128 in length, this root will be incorrect.

Reply:
makes sense 👍